packages/plugin-ext/src/main/browser/plugin-uri-handler-service.ts

```diff
@@ -0,0 +1,83 @@
+// *****************************************************************************
+// Copyright (C) 2024 STMicroelectronics.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { OpenerOptions, OpenerService, OpenHandler } from '@theia/core/lib/browser';
+import { injectable, inject } from '@theia/core/shared/inversify';
+import { Disposable, MaybePromise, URI } from '@theia/core';
+
+export interface UriHandler {
+    canHandleURI(uri: URI): boolean;
+    handleUri(uri: URI): Promise<void>;
+}
+
+export const PluginUriHandlerService = Symbol('PluginUriHandlerService');
+
+export interface PluginUriHandlerService {
+    registerUriHandler(id: string, uriHandler: UriHandler): Disposable;
+    unregisterUriHandler?(id: string): void;
+}
+
+@injectable()
+export class DefaultPluginUriHandlerService implements PluginUriHandlerService {
```

Comment:
It's not clear to me why we need this class. Can't this code just live in the `UriMainImpl` class?